apps/worker/src/app/workflow/usecases/add-job/add-job.usecase.ts

```diff
@@ -157,9 +161,38 @@ export class AddJob {
 
     const delay = this.getExecutionDelayAmount(filtered, digestAmount, delayAmount);
 
+    await this.validateDeferDuration(delay, job, command);
+
     await this.queueJob(job, delay);
   }
 
+  private async validateDeferDuration(delay: number, job: JobEntity, command: AddJobCommand) {
+    const errors = await this.tierRestrictionsValidatorUsecase.execute(
+      TierRestrictionsValidatorCommand.create({
+        deferDurationMs: delay,
+        stepType: job.type,
+        organizationId: command.organizationId,
+      })
+    );
+
+    if (errors) {
+      Logger.warn({ errors, jobId: job._id }, 'Defer duration limit exceeded for job', LOG_CONTEXT);
+
+      await this.executionLogRoute.execute(
+        ExecutionLogRouteCommand.create({
+          ...ExecutionLogRouteCommand.getDetailsFromJob(job),
+          detail: DetailEnum.DEFER_DURATION_LIMIT_EXCEEDED,
+          source: ExecutionDetailsSourceEnum.INTERNAL,
+          status: ExecutionDetailsStatusEnum.FAILED,
+          isTest: false,
+          isRetry: false,
+        })
+      );
+
+      throw new Error(DetailEnum.DEFER_DURATION_LIMIT_EXCEEDED);
```

Comment:
isn't this lacking some info, why, by how much and so on?

Reply:
maybe a specific error class for it

Reply:
this is a "safe-guard" error throw as the executionLogRoute will throw the same error inside of it. meaning we will never get to this line.

Reply:
I would make it more clear that if we got here we have a bug, so it will pop up in someones eyes if this is ever called, better that the current one